### ISSUE TITLE
Private DNS: no error while create A record with more than 20 records

### ISSUE THREAD

**Issue by amarkevich** [hashicorp/terraform-provider-azurerm#12406]
### Terraform (and AzureRM Provider) Version

Terraform v1.0.1
- Installed hashicorp/azurerm v2.66.0 (signed by HashiCorp)

### Affected Resource(s)

* `azurerm_private_dns_a_record

### Terraform Configuration Files

```hcl
resource "azurerm_private_dns_a_record" "a" {
  name                = "@"
  resource_group_name = "rg-corp-dns"
  zone_name           = "corp.com"
  ttl                 = 300
  records             = ["10.0.180.17", ...more than 20... ]
}
```

### Debug Output

```
module.corp-com-dns-zone["corp.com"].azurerm_private_dns_a_record.a["@"]: Creating...
```

### Panic Output

```
│ Error: creating/updating A Record: (A Name "@" / Private Dns Zone Name "corp.com" / Resource Group "rg-corp-dns"): privatedns.RecordSetsClient#CreateOrUpdate: Failure sending request: StatusCode=409 -- Original Error: autorest/azure: Service returned an error. Status=<nil> <nil>
│ 
│   with module.corp-com-dns-zone["corp.com"].azurerm_private_dns_a_record.a["@"],
│   on .terraform/modules/corp-com-dns-zone/main.tf line 21, in resource "azurerm_private_dns_a_record" "a":
│   21: resource "azurerm_private_dns_a_record" "a" {
```

### Expected Behaviour

Detailed error message

### Actual Behaviour

No error message

### Steps to Reproduce

1. `terraform apply`

**Comment by jackofallops** [comment]
Hi @amarkevich - I think the problem here is that you are trying to create an SOA entry as an A-record, which is not allowed by the API.  The SOA ('@') record is managed with the `azurerm_private_dns_zone` resource itself. That said, we should have some validation on the `azurerm_private_dns_a_record` resource to prevent that value being specified.

**Comment by t-l-k** [comment]
@jackofallops @amarkevich @stephybun Creating `@` type `A` records for "apex" records is distinct from type `SOA` records, no? I've raised following issue: 

- Bug https://github.com/hashicorp/terraform-provider-azurerm/issues/13081

I looked at the `soa_record` block on the resource docs and couldn't find an attribute that would accept an address. I've used `A` private DNS records in the past to set apex records up until 2.73.0, worked fine. I wasn't trying to set 20 IPs tho!
